### ISSUE TITLE
Manage document dependencies

### ISSUE THREAD

**Issue by peyerluk** [upfrontIO/livingdocs-engine#142]
## Extended Livingdoc API:

```coffee
# add js
addJsDependency(src: 'url')
addJsDependency(code: 'inline js')

# add css
livingdoc.addCssDependency(src: 'url')
livingdoc.addCssDependency(code: 'inline css')

# use namespaces
livingdoc.addJsDependency(src: 'url', namespace: 'embeds.twitter')

# Access the dependencies collection directly:
dependencies = livingdoc.dependencies

# Transform to JSON
dependencies.serialize()

# Get namespaces (Array of String)
dependencies.getNamespaces()

# Get all dependencies of a namespace
dependencies.getNamespace('embeds.twitter')

# Get an HTML string to include in a published document or to
# Add to a document on the server side.
dependencies.printJs()
dependencies.printCss()
```

## Exposed Modules:

For reuse in the editor these modules are exposed on `doc`:
`doc.JsLoader` and `doc.CssLoader`

```coffee
# Example:
jsLoader = new doc.JsLoader(window: iframe.contentWindow)
jsLoader.loadSingleUrl(url, callback)
jsLoader.loadInlineScript(url, callback)
```

**Comment by peyerluk** [comment]
```coffee
# Register a js dependency
# The name is required to identify the dependency and to
# make it easier to prevent registrations of the same dependency multiple times
livingdoc.addDependency(name: 'twitter', js: jsUrl)

# Register an inline js dependency
livingdoc.addDependency(name: 'twitter', js: jsCode, head: true, inline: true)

# Register a js dependency that is needed by a specific component
# If the component is removed the dependency is removed as well. If multiple 
# components need the same dependency it is removed if the last component
# is removed.
livingdoc.addDependency(name: 'twitter', js: jsUrl, component: component)
```

**Comment by peyerluk** [comment]
To embed the inline js and css into the document we can add a option on livingdoc for that like this:

```coffee
livingdoc.toHtml(embedInlineDependencies: true)
```

**Comment by gabrielhase** [comment]
IMO we should be more specific for the use case:
1. Embed a script
2. Call a method of this script once it's loaded
This is what we'll need in most embed cases, e.g., with Twitter. With your proposal one could do this with the inlineJS, but I think it would be nicer if we could combine the file registration with an after-load call.

**Comment by peyerluk** [comment]
Implemented in #149
